File: src/Benchmarks/Optimization/Function_Minimization/Function_Minimization.jl
include(joinpath(@__DIR__,"../../../timeit.jl"))

using Optim

function rosenbrock(x)
    D = length(x)
    sum( 100*(x[i]^2 - x[i+1])^2 + (x[i]-1)^2 for i=1:D-1 )
end

function main()
    optimize(rosenbrock,-5*ones(9),NelderMead(),Optim.Options(iterations=1500))
    true
end

@assert main()
timeit("Optimize a function with Nelder-Mead",main)

File: src/Benchmarks/Optimization/Function_Minimization/Function_Minimization.R
source("../src/timeit.R")

## benchmarks here
suppressMessages(library(neldermead))#avoid messing up our nice prints

#sum( 100*(x[i]^2 - x[i+1])^2 + (x[i]-1)^2 for i=1:D-1 )
rosenbrock = function(x){
    D = length(x)
    return(sum( 100*(x[2:D]-x[1:D-1]^2)^2 + (1-x[1:D-1])^2 ))
}

main = function(){
    opt = optimset(TolX=1.e-16, MaxIter=1500, MaxFunEvals=5000,Display="off")
    sol = fminsearch(rosenbrock, rep(-5, 9),opt)
}

timeit("Optimize a function with Nelder-Mead",main)
